vignettes/RPyGeo.Rmd

```diff
@@ -151,44 +160,29 @@ Subsequently, the polygons are converted to polylines.
 
 ### Modules and extensions
 
-To use the functions of a specific extension, you have to first access the corresponding extension via the `$` operator, and then have to use a second `$`  to get access to its functions.
-For example, to compute the slope from a digital elevation model with the help of the Spatial Analyst extenstion, we run:
+To use the functions of a specific extension, one has to enable the extension first (see `rpygeo_build_env()`. 
+To access an extension we can make use of the `$` operator.
+Then, a second `$` provides access to the functions of the extension.
+For example, to compute the slope from a DEM with the help of the Spatial Analyst extension, we can run:
 
 ```{r, eval=FALSE}
 arcpy$sa$Slope(in_raster = "elev.tif")
 ```
 
-Note that there is no output parameter, hence the result of the processing will be saved to the scratch workspace location (see again `rpygeo_build_env()`).
-
-### Load output file into R Session
-
-The output of an ArcPy function can be directly loaded into the R session with the `rpygeo_load()` function.
-
-<!-- well, I would never write that... a right-handed assignment operator-->
-```{r, eval=FALSE}
-slope <- arcpy$Slope_3d(in_raster = "elev.tif", out_raster = "slope.tif") %>%
-  rpygeo_load()
-```
-
-<!-- Marc: and is the same possible with arcpy$sa$Slope? -->
-The slope raster is loaded as a `raster` object into the R session.
-
-###  Save files
-
-Some functions have no parameter to specify an output file. 
-Instead they return an object and a temporary file is saved to the workspace.
-For example, the `sa$Aspect` function returns a `Raster` object and writes a temporary Arc/Info Binary Grid file to the workspace. 
+Note that there is no output parameter, hence the result of the processing will be saved as an Arc/Info Binary Grid file to the scratch workspace location (see again `rpygeo_build_env()`).
+To save the output to another location, we can make use of `rpygeo_save()`.
 
 ```{r, eval=FALSE}
-arcpy$sa$Aspect(in_raster("elev.tif")) %>%
-  rpygeo_save("aspect.tif")
+arcpy$sa$Slope(in_raster = "elev.tif") %>%
+  rpygeo_save("slope.tif")
 ```
 
 ### Map algebra
 
 Map algebra expressions can be used in RPyGeo with special operators. 
 The four basic calculus functions are implemented as `%rpygeo_+%`, `%rpygeo_-%`, `%rpygeo_*%`, `%rpygeo_/%`. 
 
+<!-- Marc, what kind of object is ras? I suspect it is not an r raster object, we should make this clear here, i.e. explain what arcpy$sa$Raster is doing -->
```

Comment:
Marc, what kind of object is `ras`? I suspect it is not an r raster object, we should make this clear here, i.e. explain what arcpy$sa$Raster is doing.

Reply:
`arcpy$sa$Raster` creates a Python raster object, which is linked to a temporary Esri Arc/Info Binary Grid stored in the scratch workspace. The `reticualte` package converts the Python raster object to an R object called `ras`.